### PR TITLE
ProtocolUpgrade support for data_provider

### PR DESCRIPTION
**Pull request by EdHastingsCasperAssociation**
This PR adds `ProtocolUpgrade` support to data provider. Extends #4532  
diff: https://github.com/casper-network/casper-node/compare/feat-2.0...EdHastingsCasperAssociation:feat-2.0-tc-upgrade

### CODE REVIEW

File: storage/src/data_access_layer/flush.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+}
+
+/// Represents a result of a `flush` request.
+pub enum FlushResult {
+    /// Manual sync is disabled in config settings.
+    ManualSyncDisabled,
+    /// Successfully flushed.
+    Success,
+    /// Failed to get total supply.
```
Comment (darthsiroftardis) [inline storage/src/data_access_layer/flush.rs:25]: Is this the correct comment

File: storage/src/system/protocol_upgrade.rs [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+                .borrow_mut()
+                .write(*locked_funds_period_key, value);
+        }
+        Ok(())
+    }
+
+    /// Handle bids migration.
+    pub fn handle_bids_migration(&self) -> Result<(), ProtocolUpgradeError> {
+        debug!("handle bids migration");
+        let mut borrow = self.tracking_copy.borrow_mut();
```
Comment (darthsiroftardis) [inline storage/src/system/protocol_upgrade.rs:805]: NIT: can we rename this variable

**Review by darthsiroftardis — APPROVED** [review]

File: storage/src/data_access_layer/flush.rs (hunk 1 above)
Comment (EdHastingsCasperAssociation) [reply storage/src/data_access_layer/flush.rs:25]: nope. ill fix it on the next PR

File: storage/src/system/protocol_upgrade.rs (hunk 2 above)
Comment (EdHastingsCasperAssociation) [reply storage/src/system/protocol_upgrade.rs:805]: sure...ill rename it in the next PR

**Pull request merged**